service/orchestrator/toe_test.go

```diff
@@ -0,0 +1,362 @@
+		&orchestrator.ListTargetsOfEvaluationRequest{OrderBy: "not a field"})
+	assert.Equal(t, codes.InvalidArgument, status.Code(err))
+	assert.Contains(t, err.Error(), api.ErrInvalidColumnName.Error())
+}
+
+func Test_UpdateTargetOfEvaluation(t *testing.T) {
+	var (
+		toe *orchestrator.TargetOfEvaluation
+		err error
+	)
+	orchestratorService := NewService()
+
+	// 1st case: ToE is nil
+	_, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{})
+	assert.Equal(t, codes.InvalidArgument, status.Code(err))
+
+	// 2nd case: Ids are empty
+	_, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{
+		CloudServiceId: "",
+		CatalogId:      "",
+	})
+	assert.Equal(t, codes.InvalidArgument, status.Code(err))
+
+	// 3rd case: ToE not found since there are no ToEs
+	_, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{
+		Toe: &orchestrator.TargetOfEvaluation{
+			CloudServiceId: "MyService",
+			CatalogId:      "Cat1234",
+		},
+	})
+	assert.Equal(t, codes.NotFound, status.Code(err))
+
+	// 4th case: ToE updated successfully
+	err = orchestratorService.storage.Create(&orchestrator.TargetOfEvaluation{
+		CloudServiceId: orchestratortest.MockServiceID,
+		CatalogId:      orchestratortest.MockCatalogID,
+		AssuranceLevel: &AssuranceLevelHigh,
+	})
+	assert.NoError(t, err)
+
+	// update the toe's assurance level and send the update request
+	toe, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{
+		Toe: &orchestrator.TargetOfEvaluation{
+			CloudServiceId: orchestratortest.MockServiceID,
+			CatalogId:      orchestratortest.MockCatalogID,
+			AssuranceLevel: &AssuranceLevelMedium,
+		},
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, toe)
+	assert.Equal(t, "new assurance level", toe.AssuranceLevel)
+}
+
+func Test_RemoveTargetOfEvaluation(t *testing.T) {
+	var (
+		err                             error
+		listTargetsOfEvaluationResponse *orchestrator.ListTargetsOfEvaluationResponse
+	)
+	orchestratorService := NewService()
+
+	// 1st case: Empty ToE ID error
+	_, err = orchestratorService.RemoveTargetOfEvaluation(context.Background(), &orchestrator.RemoveTargetOfEvaluationRequest{
+		CloudServiceId: "MyService",
+		CatalogId:      "Cat1234",
+	})
+	assert.Error(t, err)
+	assert.Equal(t, status.Code(err), codes.InvalidArgument)
+
+	// 2nd case: ErrRecordNotFound
+	_, err = orchestratorService.RemoveTargetOfEvaluation(context.Background(), &orchestrator.RemoveTargetOfEvaluationRequest{
+		CloudServiceId: "0000",
+		CatalogId:      "0000",
+	})
+	assert.Error(t, err)
+	assert.Equal(t, status.Code(err), codes.NotFound)
+
+	// 3rd case: Record removed successfully
+	err = orchestratorService.storage.Create(&orchestrator.TargetOfEvaluation{
```

Comment:
You need to create a catalog and cloud service first, otherwise the foreign key constraints will (correct) fail.